### PR TITLE
Medical Treatment - Change stitching action to be per-body-part

### PR DESCRIPTION
**Pull request by pterolatypus**
**When merged this pull request will:**
- Stitch each body part separately instead of all at once
- Gives medics a bit more control

### IMPORTANT

- [x] If the contribution affects [the documentation](https://github.com/acemod/ACE3/tree/master/docs), please include your changes in this pull request so the documentation will appear on the [website](https://ace3.acemod.org/).
- [x] [Development Guidelines](https://ace3.acemod.org/wiki/development/) are read, understood and applied.
- [x] Title of this PR uses our standard template `Component - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: addons/medical_treatment/functions/fnc_getStitchableWounds.sqf [hunk 1]
```diff
@@ -29,5 +30,6 @@ private _bleedingBodyParts = GET_OPEN_WOUNDS(_unit) select {
 GET_BANDAGED_WOUNDS(_unit) select {
     _x params ["", "_bodyPartN"];
 
-    !(_bodyPartN in _bleedingBodyParts)
+    (_bodyPart in [-1, _bodyPartN]) &&
+    {!(_bodyPartN in _bleedingBodyParts)}
```
Comment (mharis001) [inline addons/medical_treatment/functions/fnc_getStitchableWounds.sqf:34]:
```suggestion
    _bodyPart in [-1, _bodyPartN] && {!(_bodyPartN in _bleedingBodyParts)}
```

File: addons/medical_treatment/functions/fnc_canStitch.sqf [hunk 2]
```diff
@@ -16,6 +17,7 @@
  * Public: No
  */
 
-params ["", "_patient"];
+params ["", "_patient", "_bodyPart"];
 
-(_patient call FUNC(getStitchableWounds) isNotEqualTo [])
+private _index = ALL_BODY_PARTS find toLower _bodyPart;
+([_patient, _index] call FUNC(getStitchableWounds) isNotEqualTo [])
```
Comment (mharis001) [inline addons/medical_treatment/functions/fnc_canStitch.sqf:23]:
```suggestion
[_patient, _index] call FUNC(getStitchableWounds) isNotEqualTo []
```

**Comment by BrettMayson** [comment]
Should this be a setting? I can see this creating a lot of confusion, anger, and "bug reports" when this goes live

**Comment by veteran29** [comment]
The progress bar should say which body part is stitched IMO.

**Comment by pterolatypus** [comment]
>Should this be a setting?

I agree it might cause some confusion, but I think people will get used to it quickly enough and it's more consistent with other treatments.
Adding a setting could lead to more confusion in the long run as the interaction will be different between different servers - this is already a problem when trying to teach people how to use medical, because the behaviour can vary massively depending on the settings.

That said, I don't really care too much I just want this for my community so I'll put it in if people really want it.

>The progress bar should say which body part is stitched

I agree that could be good. Currently the progress par text is static for all medical actions, so it'd mean tweaking that and then it could be applied to virtually all treatments. I think it merits a separate PR though.

**Comment by mharis001** [comment]
I'd agree with no setting. This change makes stitching consistent with the other treatments and just seems like more natural interaction behaviour.

**Comment by kymckay** [comment]
Agree with no setting, each kit should probably get 5 uses (one for each body part) if we have a way to do that so that they're "equivalent" to before (and old loadouts are balanced as expected)

**Comment by veteran29** [comment]
If we will make surgical kit into a magazine we can easily track amount of use. Not sure if changing its type will cause any issues at this point tho.

**Comment by pterolatypus** [comment]
To be honest, consuming surgical kits is already weird with the current implementation because if you cancel before it finishes you can stitch all but 1 wound and then get your item back. I think it'd make more sense to have an item like "suturing thread" that consumes one per wound stitched.

**Comment by BrettMayson** [comment]
> To be honest, consuming surgical kits is already weird with the current implementation because if you cancel before it finishes you can stitch all but 1 wound and then get your item back. I think it'd make more sense to have an item like "suturing thread" that consumes one per wound stitched.

I believe this was the solution in the other thread this was brought up

> I'd agree with no setting

Me too, just gotta make sure it's really obvious in the changelog. We'll still have people showing up in the Discord to complain it is broken anyway though

**Comment by Drofseh** [comment]
Making a new item will require everyone to update any custom loadouts they have.

I'd prefer the existing surgical kit became multi use when consumable.

**Comment by pterolatypus** [comment]
We could always convert the existing kits

**Comment by BrettMayson** [comment]
Needs to be updated after #9020

**Review by BrettMayson — APPROVED** [review]

**Pull request merged**